pkg/controller/registry/resolver/cache.go

```diff
@@ -436,14 +436,27 @@ func WithChannel(channel string) OperatorPredicate {
 
 func WithPackage(pkg string) OperatorPredicate {
 	return func(o *Operator) bool {
+		for _, p := range o.Properties() {
```

Comment:
Shouldn't this be in `NewOperator*FromCSV`?

Reply:
If an `Operator` can have more than one `PackageType` property, then the location here makes sense; currently, we don't return more than one package from `o.Package()`.

Reply:
I didn't move this because I hope that we will be externalizing constraint filtering soon :tm: , and didn't want to touch `struct Operator` in the meantime.